mail_tracking/static/src/js/failed_message/widget.js

```diff
@@ -0,0 +1,311 @@
+/* Copyright 2019 Alexandre Díaz
+   License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html). */
+odoo.define('mail_tracking.FailedMessage', function (require) {
```

Comment:
I think that `widget.js` is not very helpful: in `discuss.js` you're altering lots of widgets too.

How about `thread.js`? Also here it could be:

```suggestion
odoo.define('mail_tracking.FailedMessageThread', function (require) {
```